intg-appletv/tv.py

```diff
@@ -514,6 +533,61 @@ async def _update_app_list(self) -> None:
 
         self.events.emit(EVENTS.UPDATE, self._device.identifier, update)
 
+    async def _update_output_devices(self) -> None:
+        _LOG.debug("[%s] Updating available output devices list", self.log_id)
+        if self._atv is None:
+            return
```

Comment:
Still not safe: `pyatv.scan` has a default timeout of 5 seconds, where the device can potentially disconnect (`connection_lost` pyatv library callback -> `self._handle_disconnect()` -> `self._atv = None`).
Therefore, the None check should be after the scan.
Unless of course my understanding of Python async is wrong :-)
I don't know if resetting `self._available_output_devices` before the None check would have a side effect or not.

Reply:
You're right, I didn't understand why the atv instance could disconnect between 2 calls